### PR TITLE
feat(cce/addon): support updating version and values

### PR DESCRIPTION
**Pull request by Jason-Zhang9309**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. update import, func name and error message
2. make the version optional
3. support updating version and values

```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/cce' TESTARGS='-run TestAccAddon_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cce -v -run TestAccAddon_basic -timeout 360m -parallel 4
=== RUN   TestAccAddon_basic
=== PAUSE TestAccAddon_basic
=== CONT  TestAccAddon_basic
--- PASS: TestAccAddon_basic (873.94s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cce       874.021s

make testacc TEST='./huaweicloud/services/acceptance/cce' TESTARGS='-run TestAccAddonTemplateDataSource_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cce -v -run TestAccAddonTemplateDataSource_basic -timeout 360m -parallel 4
=== RUN   TestAccAddonTemplateDataSource_basic
=== PAUSE TestAccAddonTemplateDataSource_basic
=== CONT  TestAccAddonTemplateDataSource_basic
--- PASS: TestAccAddonTemplateDataSource_basic (481.22s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cce       481.306s
```

### CODE REVIEW

File: huaweicloud/services/acceptance/cce/resource_huaweicloud_cce_addon_test.go [hunk 1]
```diff
@@ -224,6 +242,41 @@ data "huaweicloud_cce_addon_template" "test" {
...
+  version       = "1.25.21"
+
+  values {
+    basic       = jsondecode(data.huaweicloud_cce_addon_template.test.spec).basic
+    custom_json = jsonencode(merge(
+      jsondecode(data.huaweicloud_cce_addon_template.test.spec).parameters.custom,
+      {
+        cluster_id = huaweicloud_cce_cluster.test.id
+        tenant_id  = "%s"
+		logLevel   = 3
```
Comment (iskey) [inline huaweicloud/services/acceptance/cce/resource_huaweicloud_cce_addon_test.go]: better used spaces than tabs.

File: huaweicloud/services/acceptance/cce/resource_huaweicloud_cce_addon_test.go [hunk 2]
```diff
@@ -237,6 +290,7 @@ resource "huaweicloud_cce_addon" "test" {
       {
         cluster_id = huaweicloud_cce_cluster.test.id
         tenant_id  = "%s"
+		logLevel   = 4
```
Comment (iskey) [inline huaweicloud/services/acceptance/cce/resource_huaweicloud_cce_addon_test.go]: better used spaces than tabs.

**Review by iskey — APPROVED** [review]
/lgtm,  only some minor comments added.

File: huaweicloud/services/acceptance/cce/resource_huaweicloud_cce_addon_test.go (hunk 1 above)
Comment (Jason-Zhang9309) [reply huaweicloud/services/acceptance/cce/resource_huaweicloud_cce_addon_test.go]: fixed

File: huaweicloud/services/acceptance/cce/resource_huaweicloud_cce_addon_test.go (hunk 2 above)
Comment (Jason-Zhang9309) [reply huaweicloud/services/acceptance/cce/resource_huaweicloud_cce_addon_test.go]: fixed

**Review by iskey — APPROVED** [review]
/lgtm

File: huaweicloud/services/cce/resource_huaweicloud_cce_addon.go [hunk 3]
```diff
@@ -212,77 +208,136 @@ func resourceCCEAddonV3Create(ctx context.Context, d *schema.ResourceData, meta
...
+			},
+		},
+	}
+
+	update, err := addons.Update(cceClient, updateOpts, d.Id(), clusterID).Extract()
+	if err != nil {
+		return diag.Errorf("error updating CCEAddon: %s", err)
+	}
+
+	d.SetId(update.Metadata.Id)
```
Comment (ShiChangkuo) [inline huaweicloud/services/cce/resource_huaweicloud_cce_addon.go]: is the ID changed after updating?

Comment (Jason-Zhang9309) [reply huaweicloud/services/cce/resource_huaweicloud_cce_addon.go]: fixed

**Pull request merged**